pkg/build/api/types.go

```diff
@@ -328,6 +328,9 @@ type GitBuildSource struct {
 
 	// HTTPSProxy is a proxy used to reach the git repository over https
 	HTTPSProxy *string
+
+	// URLCheckTimeoutSeconds is the timeout in seconds to use when checking connectivity to the git server
+	URLCheckTimeoutSeconds *int
```

Comment:
How fast is fast enough for a very rare case?  1 minute?  Why make this
parametrizable when you can pick a good rule of thumb?  Do a 10s timeout,
print a message, then do 2m timeout.  That would give instant feedback,
then still handle 99% of the time.

On Mon, May 23, 2016 at 3:51 PM, Cesar Wong <notifications@github.com>
wrote:

> In pkg/build/api/types.go
> <https://github.com/openshift/origin/pull/8990#discussion_r64277168>:
>
> > @@ -328,6 +328,9 @@ type GitBuildSource struct {
> >
> >  	// HTTPSProxy is a proxy used to reach the git repository over https
> >  	HTTPSProxy *string
> > +
> > +	// URLCheckTimeoutSeconds is the timeout in seconds to use when checking connectivity to the git server
> > +	URLCheckTimeoutSeconds *int
>
> The purpose of the git ls-remote check was that we would fail fast when
> you don't have connectivity to the git URL. Otherwise git would just hang
> forever trying to do the clone.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8990/files/09988784f42909e7b7cd9619c04c27904d62d89d#r64277168>
>